digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -257,9 +283,23 @@ pub async fn get_assets_by_condition(
         .order_by(sort_by, sort_direction.clone())
         .order_by(asset::Column::Id, sort_direction);
 
+    let flag = flags.get("grand_total_flag").unwrap_or(&false).clone();
+
+    if flag {
```

Comment:
Style-wise this is acceptable but not ideal. In terms of speed/effort, it's fine to have it like this for now. But I think we can do better without much extra work.

The cleaner way of setting this up would be to create a reusable function that handles grand total. If you think about it, the grand total logic only needs the query statement + if the flag exists. Therefore you can do something like this:

```
let stmt = paginate(pagination, limit, stmt);
let (full_assets, grand_total) = get_full_asset_response(...).await?;
```

and then 

```
fn get_full_asset_response(...) -> [...] {
  let assets_task = stmt.clone().all(conn);
  if feature_flags.grand_total {
     let (assets, grand_total) = try_join!(assets_task, get_grand_total(conn, stmt.clone())).await?;
     return Ok((full_assets, Some(grand_total));
  }
  Ok(assets_task.await?, None);
}
```
And now you can use `get_full_asset_response` for getAssetsByCreator, searchAssets, etc.